content/en/docs/spin-operator/tutorials/pub-sub-with-redis.md

```diff
@@ -0,0 +1,213 @@
+├── spin.toml
+└── src
+    └── lib.rs
+```
+
+If we open the application manifest (`spin.toml` file) we see that Spin has already pre-populated the [Redis trigger configuration](https://developer.fermyon.com/spin/v2/redis-trigger#the-spin-redis-trigger):
+
+```toml
+// --snip --
+[application.trigger.redis]
+address = "redis://my-redis-master.default.svc.cluster.local:6379"
+
+[[trigger.redis]]
+channel = "channel-one"
+component = "redis-message-handler"
+// --snip --
+```
+
+By default, Spin does not authenticate to Redis. You can work around this by providing the password in the `redis://` URL. For example: `address = "redis://:p4ssw0rd@localhost:6379"`
+
+> Do not use passwords in code committed to version control systems.
+
+## Application Logic
+
+In this example, we want to write the logic for our Spin application to listen to messages published on `channel-one`. So, we open the `src/lib.rs` file and paste the following code:
+
+```rust
+use anyhow::Result;
+use bytes::Bytes;
+use spin_sdk::redis_component;
+use std::str::from_utf8;
+
+/// A simple Spin Redis component.
+#[redis_component]
+fn on_message(message: Bytes) -> Result<()> {
+    println!("{}", from_utf8(&message)?);
+    Ok(())
+}
+```
+
+## Build
+
+With the logic and configuration in place, we can build the Spin application:
+
+```bash
+spin build
+```
+
+## Publish
+
+We will now push the application image to a registry. You can use any container registry you prefer (like DockerHub). But for this tutorial, we’ll use a simple one that does not require authentication:
+
+```bash
+spin registry push ttl.sh/redis-message-handler:0.1.0
+```
+
+To read the configuration we can use the `spin scaffold` command:
+
+```bash
+spin kube scaffold --from ttl.sh/redis-message-handler:0.1.0
+```
+
+As we can see, our `SpinApp` is all set and using the `containerd-shim-spin` executor:
+
+```yaml
+apiVersion: core.spinoperator.dev/v1alpha1
+kind: SpinApp
+metadata:
+  name: redis-message-handler
+spec:
+  image: "ttl.sh/redis-message-handler:0.1.0"
+  executor: containerd-shim-spin
+  replicas: 2
+```
+
+## Deploy
+
+We deploy our application to our cluster using the `spin kube` command:
+
+```bash
+spin kube deploy --from ttl.sh/redis-message-handler:0.1.0
+```
+
+## Test
+
+We want to run the Redis server and publish a message. First, we run the Redis container image:
+
+```bash
+kubectl run --namespace default redis-client --restart='Never'  --env REDIS_PASSWORD=$REDIS_PASSWORD  --image docker.io/bitnami/redis:7.2.4-debian-12-r9 --command -- sleep infinity
+```
+
+Then, we want to attach to the pod:
+
+```bash
+kubectl exec --tty -i redis-client --namespace default -- bash
+```
+
+And, access the Redis CLI from inside the cluster:
+
+```bash
+REDISCLI_AUTH="$REDIS_PASSWORD" redis-cli -h my-redis-master
+```
+
+Which then provides us with the prompt (`my-redis-master:6379>`) at which point we can publish our message:
+
+```bash
+my-redis-master:6379> PUBLISH channel-one message-one
+```
```

Comment:
Lastly, shall we show the redis app pod logs printing receipt of the message?

Reply:
Hi @vdice 
I made it all the way to the end of this doc on my machine, but am getting `integer 0` when I publish the message on channel-one.
I feel like I am missing something; should I be writing any Rust code after the `// Implement me` comment?

Reply:
Hi @tpmccallum, that's all good on the publishing side.  What I meant here was to then show the logs from the consuming side, eg the Spin app that is subscribing to messages published to the `channel-one` channel.  For example, here are the logs for a redis SpinAapp when I published "HELLO" to the `messages` channel:

```
$ kubectl logs -f -l core.spinoperator.dev/app-name=redis-spinapp
Active Channels on redis://:DTatx2DV4p@redis-master.redis.svc.cluster.local:6379:
	messages: [redis-rust]
HELLO
```

Reply:
Ah, nice one @vdice!